frontend/pages/policies/ManagePoliciesPage/ManagePoliciesPage.tsx

```diff
@@ -394,14 +400,21 @@ const ManagePolicyPage = ({
 
   // NOTE: used to reset page number to 0 when modifying filters
   const handleResetPageIndex = () => {
+    // this function encapsulates setting local page state to 0 and triggering the descendant
+    // TableContainer to do the same via resetPageIndex – see comment above that state definition.
     setTableQueryDataForApi(
       (prevState) =>
         ({
           ...prevState,
           pageIndex: 0,
         } as ITableQueryData)
     );
+    // change in state triggers effect in TableContainer (see comment above this state definition)
     setResetPageIndex(true);
+    // timeout ensures effect in TableContainer runs before setting it back to false here. Without
+    // timeout, the effect doesn't see a change.
+    setTimeout(() => setResetPageIndex(false), 1);
```

Comment:
A fix requiring `setTimeout()` is a red flag for me.  A strategy I've used before is to set `resetPageIndex` to a number rather than a bool, and increment it every time you need the child component to reset.

Reply:
That's a good idea, will try it out here